sdk/resourcemanager/azure-resourcemanager-resources/src/main/java/com/azure/resourcemanager/resources/fluentcore/arm/Manager.java

```diff
@@ -56,6 +58,10 @@ protected final void withResourceManager(ResourceManager resourceManager) {
         this.resourceManager = resourceManager;
     }
 
+    protected void withInternalHttpPipeline(AzureConfigurable<?> azureConfigurable, HttpPipeline httpPipeline) {
```

Comment:
Do we need `protected`, or `private` would do?

Reply:
The `protected` would help reuse this method for different managers.

Reply:
better return `AzureConfigurable` for chaining like `withInternalPipeline().authenticate()`